js/mexc3.js

```diff
@@ -0,0 +1,3659 @@
+        //         "data":{
+        //             "currency":"USDC",
+        //             "chains":[
+        //                 {"chain":"ERC-20","address":"0x55cbd73db24eafcca97369e3f2db74b2490586e6"},
+        //                 {"chain":"MATIC","address":"0x05aa3236f1970eae0f8feb17ec19435b39574d74"},
+        //                 {"chain":"TRC20","address":"TGaPfhW41EXD3sAfs1grLF6DKfugfqANNw"},
+        //                 {"chain":"SOL","address":"5FSpUKuh2gjw4mF89T2e7sEjzUA1SkRKjBChFqP43KhV"},
+        //                 {"chain":"ALGO","address":"B3XTZND2JJTSYR7R2TQVCUDT4QSSYVAIZYDPWVBX34DGAYATBU3AUV43VU"}
+        //             ]
+        //         }
+        //     }
+        //
+        const data = this.safeValue (response, 'data', {});
+        const chains = this.safeValue (data, 'chains', []);
+        const depositAddresses = [];
+        for (let i = 0; i < chains.length; i++) {
+            const depositAddress = this.parseDepositAddress (chains[i], currency);
+            depositAddresses.push (depositAddress);
+        }
+        return this.indexBy (depositAddresses, 'network');
+    }
+
+    async fetchDepositAddress (code, params = {}) {
+        const rawNetwork = this.safeStringUpper (params, 'network');
+        params = this.omit (params, 'network');
+        const response = await this.fetchDepositAddressesByNetwork (code, params);
+        const networks = this.safeValue (this.options, 'networks', {});
+        const network = this.safeString (networks, rawNetwork, rawNetwork);
+        let result = undefined;
+        if (network === undefined) {
+            result = this.safeValue (response, code);
+            if (result === undefined) {
+                const alias = this.safeString (networks, code, code);
+                result = this.safeValue (response, alias);
+                if (result === undefined) {
+                    const defaultNetwork = this.safeString (this.options, 'defaultNetwork', 'ERC20');
+                    result = this.safeValue (response, defaultNetwork);
+                    if (result === undefined) {
+                        const values = Object.values (response);
+                        result = this.safeValue (values, 0);
+                        if (result === undefined) {
+                            throw new InvalidAddress (this.id + ' fetchDepositAddress() cannot find deposit address for ' + code);
+                        }
+                    }
+                }
+            }
+            return result;
+        }
+        // TODO: add support for all aliases here
+        result = this.safeValue (response, rawNetwork);
+        if (result === undefined) {
+            throw new InvalidAddress (this.id + ' fetchDepositAddress() cannot find ' + network + ' deposit address for ' + code);
+        }
+        return result;
+    }
+
+    async fetchDeposits (code = undefined, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
```

Comment:
From my tests, I don't think this is working as is. When we provide a code let's say USDT we have to attach the network to it (although I did not find any mention in the docs regarding that).

So a valid request would be

```
request['currency'] = 'USDT-TRX';

```
Please check this output:

```
 node cli.js mexc3 fetchDeposits "USDT" --verbose
Debugger attached.
2022-04-22T16:47:44.066Z
Node.js: v16.13.2
CCXT v1.77.63
mexc3.fetchDeposits (USDT)
fetch Request:
 mexc3 GET https://www.mexc.com/open/api/v2/asset/deposit/list?currency=USDT-TRX 
RequestHeaders:

RequestBody:
 undefined 

handleRestResponse:
 mexc3 GET https://www.mexc.com/open/api/v2/asset/deposit/list?currency=USDT-TRX 200 OK 
ResponseHeaders:
ResponseBody:
 {"code":200,"data":{"page_size":20,"total_page":1,"total_size":2,"page_num":1,"result_list":[{"currency":"USDT-TRX","amount":61.984632,"fee":0.0,"confirmations":40,"address":"TGKyBGx7m3QGNJctddF1eEr7eVMJQB3mfC","state":"SUCCESS","tx_id":"12b9bcd2aefe52f686bc0a3e73bc211c699f8f47aedcd774252dee4a05475870","require_confirmations":20,"create_time":"2022-04-22T15:52:59.000+00:00","update_time":"2022-04-22T15:53:00.000+00:00"},{"currency":"USDT-TRX","amount":150.0,"fee":0.0,"confirmations":40,"address":"TGKyBGx7m3QGNJctddF1eEr7eVMJQB3mfC","state":"SUCCESS","tx_id":"0438e31941fcfaada5ff91d3880485135390d7dc19d459b0709deeba908a5644","require_confirmations":20,"create_time":"2022-04-22T09:22:59.000+00:00","update_time":"2022-04-22T09:23:00.000+00:00"}]}} 

2022-04-22T16:47:51.132Z iteration 0 passed in 385 ms

id |                                                             txid |     timestamp |                 datetime | network |                            address | addressTo | addressFrom | tag | tagTo | tagFrom |    type |    amount | currency | status |       updated |                          fee
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
   | 0438e31941fcfaada5ff91d3880485135390d7dc19d459b0709deeba908a5644 | 1650619379000 | 2022-04-22T09:22:59.000Z |   TRC20 | TGKyBGx7m3QGNJctddF1eEr7eVMJQB3mfC |           |             |     |       |         | deposit |       150 |     USDT |     ok | 1650619380000 | {"cost":0,"currency":"USDT"}
   | 12b9bcd2aefe52f686bc0a3e73bc211c699f8f47aedcd774252dee4a05475870 | 1650642779000 | 2022-04-22T15:52:59.000Z |   TRC20 | TGKyBGx7m3QGNJctddF1eEr7eVMJQB3mfC |           |             |     |       |         | deposit | 61.984632 |     USDT |     ok | 1650642780000 | {"cost":0,"currency":"USDT"}
2 objects
Waiting for the debugger to disconnect...
```

Reply:
This aspect of networks is not entirely unified yet, check huobi.js, okx.js, mexc.js (v1) for reference.
In short, the unified `currency` structure must have a `networks` subdictionary. The networks codes should be translated to exchange-specific network names and back in methods like withdraw, fetchWithdrawals, fetchDeposits or fetchDepositAddress and related parsing methods. We also have a more or less unified `network` param. And some unified network identifiers. This is still a work in progress (in its early stage).

Reply:
also one note about deposits - i've explained that in telegram chat - they have different chain names for same (i.e. bsc) network accross different coins, so 'fetchDeposits' might misbehave if you will append same chain name to all i.e. bsc networks, like:
`coinname-BSC` as each coin might need to save their specific chain-title in instance property, and extract those chain-titles inside `fetchDeposits` for that specific coin name... shortly, a bit tricky thing.